src/components/templates/PartyMap/components/Map/PartyMapRoomOverlay.tsx

```diff
@@ -14,104 +17,205 @@ import RoomAttendance from "../RoomAttendance";
 interface PartyMapRoomOverlayProps {
   // Passed down from Map component
   room: PartyMapRoomData;
+  enterRoom: (room: PartyMapRoomData) => void;
 
   // Passed down from Camp component (via Map component)
   venue: PartyMapVenue;
   attendances: Attendances;
   selectedRoom?: PartyMapRoomData;
-  setSelectedRoom: (room?: PartyMapRoomData) => void;
-  setIsRoomModalOpen: (value: boolean) => void;
+  selectRoom: (room: CampRoomData) => void;
+  unselectRoom: () => void;
 }
 
 export const PartyMapRoomOverlay: React.FC<PartyMapRoomOverlayProps> = ({
   room,
   venue,
   attendances,
   selectedRoom,
-  setSelectedRoom,
-  setIsRoomModalOpen,
+  selectRoom,
+  unselectRoom,
 }) => {
   const isSelectedRoom = room === selectedRoom;
+
+  // TODO: can we cleanup/de-dupe these attendance/attendees things?
+  const attendees =
+    (attendances[`${venue.name}/${room.title}`] ?? 0) +
+    (room.attendanceBoost ?? 0);
   const hasAttendance = attendances[`${venue.name}/${room.title}`];
+  const hasAttendees = attendees > 0;
 
-  // TODO: Pass this down from Camp, maybe called selectRoom?
-  const selectRoom = useCallback(
-    (room: PartyMapRoomData) => {
-      setSelectedRoom(room);
-      setIsRoomModalOpen(true);
-    },
-    [setIsRoomModalOpen, setSelectedRoom]
-  );
+  // const isRoomExternal = isExternalUrl(room.url);
+  // const currentRoomUrl = getRoomUrl(room.url);
 
-  const toggleRoom = useCallback(() => {
+  // TODO: do we still need this?
+  // const [roomClicked, setRoomClicked] = useState<string | undefined>(undefined);
+  // TODO: do we need the whole room object? Can we just roll this logic into whatever is setting roomClicked, then rename above to isRoomClicked?
+  // const isThisRoomClicked = roomClicked === room.title;
+
+  // TODO: I think this may be intended to toggle 'clicked' for a room, do we need to separately keep track of 'clicked'? Or is it the same as selecting the room?
+  // TODO: what is a better semantic name for this?
+  const onClick1 = useCallback(() => {
+    // if (!IS_BURN) {
     selectRoom(room);
+    // } else {
+    // setRoomClicked((prevRoomClicked) =>
+    //   prevRoomClicked === room.title ? undefined : room.title
+    // );
+    // }
   }, [room, selectRoom]);
 
-  const [isRoomHovered, setRoomHovered] = useState<
-    PartyMapRoomData | undefined
-  >(undefined);
+  // TODO: this was used in place of onClick1 in PartyMap Map before I merged them, is it correct? Or is the onClick1 definition above?
+  // const toggleRoom = useCallback(() => {
+  //   selectRoom(room);
+  // }, [room, selectRoom]);
```

Comment:
- [ ] [TODO] this was used in place of `onClick1` in `PartyMap` `Map` before I merged them, is it correct? Or is the `onClick1` definition above?

Reply:
resolved in https://github.com/sparkletown/sparkle/pull/840/commits/4b6a1669636861e45d8f7344d9ba82a8437b29c8